src/models/subscription.ts

```diff
@@ -0,0 +1,197 @@
+  installationId?: number;
+  jiraHost?: string;
+  numberOfSyncedRepos?: number;
+  repos?: Repositories;
+}
+
+interface SyncStatusCount {
+  syncStatus: string;
+  count: number;
+}
+
+export interface Repositories {
+  [id: string]: {
+    repository?: Repository;
+    pullStatus?: string;
+    branchStatus?: string;
+    commitStatus?: string;
+    // TODO: need to get concrete typing
+    [key: string]: unknown;
+  };
+}
+
+export interface Repository {
+  id: string;
+  name: string;
+  full_name: string;
+  owner: { login: string };
+  html_url: string;
+  updated_at: number; // TODO: is this a date object or a timestamp?  Different places uses different things
+}
+
+export default class Subscription extends Sequelize.Model {
+  gitHubInstallationId: number;
+  jiraHost: string;
+  selectedRepositories: number[];
+  repoSyncState: RepoSyncState;
+  syncStatus: SyncStatus;
+  syncWarning: string;
+  jiraClientKey: string;
+  updatedAt: Date;
+  createdAt: Date;
+
+  static async getAllForHost(host: string): Promise<Subscription[]> {
+    return Subscription.findAll({
+      where: {
+        jiraHost: host,
+      },
+    });
+  }
+
+  static getAllForInstallation(
+    installationId: number,
+  ): Promise<Subscription[]> {
+    return Subscription.findAll({
+      where: {
+        gitHubInstallationId: installationId,
+      },
+    });
+  }
+
+  static getAllForClientKey(clientKey: string): Promise<Subscription[]> {
+    return Subscription.findAll({
+      where: {
+        jiraClientKey: clientKey,
+      },
+    });
+  }
+
+  static getSingleInstallation(
+    jiraHost: string,
+    gitHubInstallationId: number,
+  ): Promise<Subscription> {
+    return Subscription.findOne({
+      where: {
+        jiraHost,
+        gitHubInstallationId,
+      },
+    });
+  }
+
+  static async getInstallationForClientKey(
+    clientKey: string,
+    installationId: string,
+  ): Promise<Subscription> {
+    return Subscription.findOne({
+      where: {
+        jiraClientKey: clientKey,
+        gitHubInstallationId: installationId,
+      },
+    });
+  }
+
+  static async install(payload: SubscriptionPayload): Promise<Subscription> {
+    const [subscription] = await Subscription.findOrCreate({
+      where: {
+        gitHubInstallationId: payload.installationId,
+        jiraHost: payload.host,
+        jiraClientKey: payload.clientKey,
+      },
+    });
+
+    await Subscription.findOrStartSync(subscription);
+
+    return subscription;
+  }
+
+  static async uninstall(payload: SubscriptionPayload): Promise<void> {
+    await Subscription.destroy({
+      where: {
+        gitHubInstallationId: payload.installationId,
+        jiraHost: payload.host,
+      },
+    });
+  }
+
+  static async findOrStartSync(
+    subscription: Subscription,
+    syncType?: string,
+  ): Promise<Job> {
+    const { gitHubInstallationId: installationId, jiraHost } = subscription;
+
+    const repoSyncState = subscription.get('repoSyncState');
```

Comment:
Not sure what's going on but stuff doesn't seem to be syncing properly for me. Sometimes I get the 'PENDING' status, others times it shows 'IN PROGRESS' but never completes (just changes to 0 repos synced... although I can still work with the repo that I selected).

![Screen Shot 2021-06-08 at 10 05 08 am](https://user-images.githubusercontent.com/37155488/121104730-fe51bb80-c845-11eb-889f-73c49691ddf4.png)

![Screen Shot 2021-06-08 at 10 05 22 am](https://user-images.githubusercontent.com/37155488/121104770-0e699b00-c846-11eb-82c3-3d8a780ac9f0.png)

Haven't had time to look into the issue coz I've gotta run but I flagged this line because `repoSyncState` appears to be undefined.

![Uploading Screen Shot 2021-06-08 at 10.34.45 am.png…]()

Reply:
I had the effect of always getting the `PENDING` state in Micros when the worker wasn't running to process items from the queues. Are you running the worker process? `npm run start` should start it, though.

Reply:
Both these scenarios (PENDING and getting stuck on IN PROGRESS) happened when running npm start

Reply:
<img width="396" alt="Screen Shot 2021-06-08 at 2 45 51 pm" src="https://user-images.githubusercontent.com/58801956/121124420-3a961380-c868-11eb-8d48-1f65ce7402de.png">

Same here, it's always `IN PROGRESS` with 0 repos synced, but the integration is working with the repo that I selected.